### DISCUSSION TITLE
Not able to run examples with ssr and cargo-leptos

### DISCUSSION THREAD

**Discussion by Grafcube** [leptos-rs/leptos#866] [Q&A]
I tried to run both the tailwind example and the ssr modes example. Just running `cargo leptos watch` works but using ssr or hydration gives me this error:

```
could not compile `mio` (lib) due to 44 previous errors
```

Am I missing a package? I have no idea how to fix this. Any help is appreciated.

**Comment by Grafcube** [comment]
Nevermind I completely misunderstood how cargo-leptos works :facepalm:

**Comment by jrasanen** [comment]
How did you solve this? What was the solution? I'm trying to run my project via `cargo leptos watch` and keep getting the error `error: could not compile `mio` (lib) due to 47 previous errors`. I'm only using SSR, no hydration or CSR.

**Reply by dustypomerleau** [reply]
`mio` errors are very frequently the result of trying to compile something to wasm that can't be compiled to wasm. The usual cause is that you have some code that's only meant to run on the server, and that code uses libraries that can't be compiled for a wasm target, but you haven't gated the code with `#[cfg(feature = "ssr")]`, so `cargo-leptos` is trying to compile it for both targets.

You can see examples of making deps optional and feature-gated in the [Axum template `Cargo.toml`](https://github.com/leptos-rs/start-axum/blob/main/Cargo.toml).

**Reply by jrasanen** [reply]
Thanks,

I think my issue is that Leptos isn't able to run without a front-end package, and I have to provide it at least something as the library. :/

When I run my server via Leptos watch, I get the following,

```
> cargo leptos watch
Error: at `C:\Users\jussi\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cargo-leptos-0.2.20\src/lib.rs:44:76`

Caused by:
    0: at `C:\Users\jussi\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cargo-leptos-0.2.20\src\config\mod.rs:59:84`
    1: missing field `lib-package`
```

I'm unsure what I would provide it as the library, but if I provide the crate that contains the Leptos example app, I get the mio errors.

And if I run the whole app via Cargo run, I get

```
2024-10-27T10:39:05.709397Z DEBUG sqlx::query: summary="COMMIT" db.statement="" rows_affected=0 rows_returned=0 elapsed=151.7µs elapsed_secs=0.0001517
It looks like you're trying to compile Leptos without the LEPTOS_OUTPUT_NAME environment variable being set. There are two options
 1. cargo-leptos is not being used, but get_configuration() is being passed None. This needs to be changed to Some("Cargo.toml")
 2. You are compiling Leptos without LEPTOS_OUTPUT_NAME being set with cargo-leptos. This shouldn't be possible!
2024-10-27T10:39:05.718374Z  INFO server: Listening on http://127.0.0.1:3000    
```

Currently, I have a `server` crate that contains as the executable binary, and a `presentation` crate that provides a single function for the Axum binary: `fn create_router(state: AppState) -> Router { ... }`. The presentation crate handles axum routing and sets up Askama templates for specific URLs. I’d like to replace the Askama templates with Leptos for SSR, but it seems that I also need to set up a WASM build to make this work ..or something.

**Reply by gbj** [reply]
@jrasanen You do not need to compile and run a frontend package if you're just doing server-side templating.

`cargo-leptos` exists only to coordinate between client and server builds, and so it is not useful for you.

If you read the text of the message you posted, `1.` gives you the solution:
```
It looks like you're trying to compile Leptos without the LEPTOS_OUTPUT_NAME environment variable being set. There are two options
 1. cargo-leptos is not being used, but get_configuration() is being passed None. This needs to be changed to Some("Cargo.toml")
```
In your `main.rs`, where it calls `get_configuration(None)` you want `get_configuration(Some("Cargo.toml"))` instead.

You can also simply use `leptos` as a library; i.e., add the crate and use the functionality. I'd suggest using 0.7 as it is better suited for this standalone purpose: you can simply `view! { <p>Hello!</p> }.into_html()` and have a `String`, no extra scaffolding required.

**Reply by jrasanen** [reply]
@gbj, thank you for responding! I've spent the whole day looking into Leptos, GitHub examples, documentation and the book are all fantastic resources.

I eventually did get it working like in your suggestion. I'll definitely take a look at version 0.7. 👍
